python.d/go_expvar.chart.py

```diff
@@ -0,0 +1,229 @@
+                self.info("Chart {} has no ID or no lines defined, skipping".format(chart))
+                continue
+
+            chart_dict['options'] = [
+                chart_opts.get('name', ''),
+                chart_opts.get('title', ''),
+                chart_opts.get('units', ''),
+                chart_opts.get('family', ''),
+                chart_opts.get('context', ''),
+                chart_opts.get('chart_type', 'line')
+            ]
+            chart_dict['lines'] = list()
+
+            # add the lines to the chart
+            for line in chart_lines:
+
+                ev_key = line.get('expvar_key')
+                ev_type = line.get('expvar_type')
+                line_id = line.get('id')
+
+                if not all([ev_key, ev_type, line_id]):
+                    self.info('Line missing expvar_key, expvar_type, or line_id, skipping: {}'.format(line))
+                    continue
+
+                if ev_type not in ['int', 'float']:
+                    self.info('Unsupported expvar_type "{}". Must be "int" or "float"'.format(ev_type))
+                    continue
+
+                if ev_key in self.expvars:
+                    self.info("Duplicate expvar key {}: skipping line.".format(ev_key))
+                    continue
+
+                self.expvars[ev_key] = ev_type
+
+                chart_dict['lines'].append(
+                    [
+                        line.get('id', ''),
+                        line.get('name', ''),
+                        line.get('algorithm', ''),
+                        line.get('multiplier', 1),
+                        line.get('divisor', 100 if ev_type == 'float' else 1),
+                        line.get('hidden', False)
+                    ]
+                )
+
+            self.order.append(chart_id)
+            self.definitions[chart_id] = chart_dict
+
+    def _get_data(self):
+        """
+        Format data received from http request
+        :return: dict
+        """
+
+        data = json.loads(self._get_raw_data())
+
+        expvars = dict()
+        if self.configuration.get('collect_memstats'):
+            expvars.update(self._parse_memstats(data))
+
+        if self.configuration.get('extra_charts'):
+            del (data['memstats'])
+            flattened = flatten(data)
+            for k, v in flattened.items():
+                key_type = self.expvars.get(k)
+                try:
+                    if key_type == 'int':
+                        expvars[k] = int(v)
+                    elif key_type == 'float':
+                        # if the value type is float, multiply it by 1000 and set line divisor to 1000
+                        expvars[k] = float(v) * 100
+                    else:
+                        # the key_type is None (not defined in config)
+                        continue
+                except ValueError:
+                    self.info("Failed to parse value for key {} as {}, ignoring key.".format(k, key_type))
```

Comment:
Python 2.6.6 (r266:84292, Aug 12 2014, 07:57:07) 
>>> '{}'.format(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: zero length field name in format


>>> '{0}'.format(1)
'1'